plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -640,19 +640,80 @@ public function get_total_shipping_refunded( $order ) {
 		return 0;
 	}
 
+	/**
+	 * Finds an Order ID based on an order key.
+	 *
+	 * @param string $order_key An order key has generated by.
+	 * @return int The ID of an order, or 0 if the order could not be found
+	 */
 	public function get_order_id_by_order_key( $order_key ) {
-		// TODO: Implement get_order_id_by_order_key() method.
-		return 0;
+		global $wpdb;
+
+		$orders_table = self::get_orders_table_name();
+		$op_table     = self::get_operational_data_table_name();
+
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		return $wpdb->get_var(
+			$wpdb->prepare(
+				"SELECT {$orders_table}.id FROM {$orders_table}
+				INNER JOIN {$op_table} ON {$op_table}.order_id = {$orders_table}.id
```

Comment:
Just noting that this `JOIN` is a bit superfluous because we could just extract `order_id` from the operational data table, but I was thinking it adds an additional protection layer by making sure the resulting ID has a corresponding record in the orders table. Whether this is really necessary or not... I'm not entirely sure.

Reply:
Yep that sounds reasonable, especially since we don't have (/can't use) foreign key constraints—and we could always revise later if a case is made for stripping it down in the name of performance.